bot.py

```diff
@@ -0,0 +1,328 @@
+        selectionScreen_fs = self.upload_to_dms(selectionscreen_name)
+        self.bot_output.add_variable(
+            key=self.input.selectionScreenVariableName.value,
+            val=selectionScreen_fs,
+        )
+        time.sleep(5)
+
+    def final_screen(self):
+        self.sapGui.findById("wnd[0]/mbar/menu[0]/menu[2]").select()
+        time.sleep(2)
+        self.sapGui.findById("wnd[1]/tbar[0]/btn[13]").press()
+        time.sleep(2)
+        self.sapGui.findById("wnd[1]/usr/btnSOFORT_PUSH").press()
+        time.sleep(2)
+        self.sapGui.findById("wnd[1]/tbar[0]/btn[11]").press()
+        time.sleep(2)
+        self.sapGui.findById("wnd[0]/mbar/menu[2]/menu[0]/menu[1]").select()
+        self.sapGui.findById("wnd[1]/usr/txtV-LOW").text = self.input.variant.value
+        self.sapGui.findById("wnd[1]/usr/txtV-LOW").caretPosition = 14
+        self.sapGui.findById("wnd[1]/tbar[0]/btn[8]").press()
+        time.sleep(2)
+        reportscreenfile_name = self.input.reportscreen_filename.value
+        time.sleep(5)
+        self.take_screenshot(reportscreenfile_name)
+        DistributionScreen_fs = self.upload_to_dms(reportscreenfile_name)
+        self.bot_output.add_variable(
+            key=self.input.reportscreenScreenshotVariableName.value,
+            val=DistributionScreen_fs,
+        )
+        container_id = "wnd[0]/usr/cntlALV_CONTAINER_3/shellcont/shell"
+        menu_item_id = "&XXL"
+        control = self.sapGui.findById(container_id)
+        control.contextMenu()
+        control.selectContextMenuItem(menu_item_id)
+        self.sapGui.findById("wnd[1]/tbar[0]/btn[0]").press()
+        self.sapGui.findById("wnd[1]/usr/ctxtDY_PATH").text = self.working_dir
+        self.sapGui.findById("wnd[1]/usr/ctxtDY_FILENAME").text = self.input.filename.value
+        self.sapGui.findById("wnd[1]/usr/ctxtDY_FILENAME").caretPosition = 18
+        self.sapGui.findById("wnd[1]/tbar[0]/btn[0]").press()
+        self.outputs.reportfilesignature.value = self._dms.upload_file_to_dms(self.input.filename.value)
+        self.log.info(message="This report file has been uploaded to DMS")
+        self.close_excel()
+
+
+    def close_excel(self):
+        try:
+            for process in psutil.process_iter(['pid', 'name']):
+                if process.info['name'] == 'EXCEL.EXE':
+                    process.terminate()
+                    process.wait(timeout=5)
+            self.log.info(message="Excel application closed")
+        except Exception as e:
+            self.log.error(f"Error closing Excel application: {e}")
+            raise
+
+
+    def log_off_sap(self):
+        try:
+            self.sapGui.findById("wnd[0]/tbar[0]/okcd").text = "/nex"
+            self.sapGui.findById("wnd[0]").sendVKey(0)
+            self.log.info(message="Logged off from SAP")
+        except Exception as e:
+            self.log.error(f"Error logging off from SAP: {e}")
+            self.bot_output.error()
```

Comment:
bot_output.error() exist only once in a entire bot logic. Please check this

Reply:
done na